indico/web/client/js/custom_elements/ind_share_widget.js

```diff
@@ -0,0 +1,362 @@
+            <Icon name="share square" styleName="icon" />
+            <HeaderContent styleName="title">
+              <Translate>Share on social media</Translate>
+            </HeaderContent>
+          </Header>
+          <Grid columns={2} stretched>
+            {!_.isEmpty(Indico.User) ? (
+              <SocialButton
+                name="Mastodon"
+                logo={`${Indico.Urls.ImagesBase}/mastodon.svg`}
+                color="violet"
+                render={button => (
+                  <SetupMastodonServer
+                    button={button}
+                    setMastodonOpen={setMastodonOpen}
+                    urlParams={shareUrlText(eventName, eventStartDt, eventUrl)}
+                  />
+                )}
+                button
+                buttonProps={{
+                  as: 'a',
+                  target: '_blank',
+                  href: Indico.User.mastodonServerURL
+                    ? `${Indico.User.mastodonServerURL}/share?text=${eventName} (${eventStartDt}) · Indico (${eventUrl})`
+                    : null,
+                }}
+              />
+            ) : (
+              <SocialButton
+                name="Mastodon"
+                logo={`${Indico.Urls.ImagesBase}/mastodon.svg`}
+                color="grey"
+                render={button => (
+                  <Popup
+                    trigger={button}
+                    content={Translate.string('Please login to share on Mastodon')}
+                    position="top right"
+                    wide
+                  />
+                )}
+                button
+              />
+            )}
+            <SocialButton
+              name="Twitter"
+              logo={`${Indico.Urls.ImagesBase}/twitter.svg`}
+              url={`https://twitter.com/intent/tweet?text=${eventName} (${eventStartDt}) · Indico (${eventUrl})`}
+              color="blue"
+              render={button => button}
+            />
+          </Grid>
+        </>
+      }
+      on="click"
+      onOpen={() => setIsPopupOpen(true)}
+      onClose={() => !isMastodonOpen && setIsPopupOpen(false)}
+      position="top right"
+      pinned
+      open={isPopupOpen}
+      wide="very"
+      style={{width: '450px'}}
+    />
+  );
+}
+
+customElements.define(
+  'ind-share-widget',
+  class extends HTMLElement {
+    connectedCallback() {
+      ReactDOM.render(
+        <ShareWidget
+          eventName={this.getAttribute('event-name')}
+          eventStartDt={this.getAttribute('event-start-dt')}
+          eventUrl={this.getAttribute('event-url')}
+          shareIcon={this.getAttribute('share-icon')}
+          googleCalParams={this.getAttribute('google-cal-params')}
+          outlookCalParams={this.getAttribute('outlook-cal-params')}
+        />,
+        this
+      );
+    }
+
+    disconnectedCallback() {
+      ReactDOM.unmountComponentAtNode(this);
+    }
+  }
+);
+
+ShareWidget.propTypes = {
```

Comment:
I'd move that right after the definition of `ShareWidget` and keep the custom element at the end